extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSink.java

```diff
@@ -47,13 +47,17 @@ private S3DataSink() {
     protected StreamResult<Object> transferParts(List<DataSource.Part> parts) {
         for (var part : parts) {
             try (var input = part.openStream()) {
+                String keyName = this.keyName;
+                if (parts.size() != 1) {
+                    keyName = this.keyName + "/" + part.name();
```

Comment:
could we avoid local variable reassigment? a ternary operator would handle this nicely

Reply:
Also commented at @dorkau80 s #156. As the `name` method/attribute within `DataSource.Part` might hold valuable metadata, it would be advisable to leverage it for uniform object naming in the destination. While `S3DataSink` could hold its own `keyPrefix`, but that as a strategy to prevent collisions in the destination bucket.

Reply:
It makes sense; however, when we initiate the transfer, we do not know exactly what the asset will be – whether a single or multiple objects. Therefore, I assumed it should work in the case of multiple objects as a bundler name to avoid collisions.

Reply:
Exactly, when we start the transfer we do not know if the asset was registered as single blob or multi blobs - that is why I use in DataSink keyPrefix check from DataSource.

Reply:
> It makes sense; however, when we initiate the transfer, we do not know exactly what the asset will be – whether a single or multiple objects. Therefore, I assumed it should work in the case of multiple objects as a bundler name to avoid collisions.

> Exactly, when we start the transfer we do not know if the asset was registered as single blob or multi blobs - that is why I use in DataSink keyPrefix check from DataSource.

Do we need the stream size? 
In `S3DataSink`, where isolated actions occur for each `DataSource.Part`, stream size is irrelevant. Given S3 bucket behavior, it makes sense to allow the `Consumer` to bundle objects in their destination, potentially containing other relevant items, however, clarifying my earlier point, it's important not to repurpose the `keyName` property, as doing so would compromise its intended semantics. This is why I suggested that `S3DataSink` could introduce its own `keyPrefix`.

Considering this, the `Consumer` needs to be mindful of potential key collisions in their S3 bucket. To mitigate this, they can take proactive measures, such as allocating a separate bucket for the transfer or utilizing the collision prevention strategy we provide (`keyPrefix` on the `DataSink` side). Implementing such measures would render the `keyName` obsolete in `S3DataSink.

Reply:
@yurimssilva Do I understand correctly - that your suggestion is to completely not use the keyName in DataSink when it comes to the name for storing transferred object, no matter if this is single or multi objects transfer? that we should always used the original part name?  

If we are on the other hand talking about the bucket name conflicts or having a possibility to store transferred objects in particular folder - that can be easily implemented - by keyPrefix (if it is not confusing) or by another property - like for example we did it for Azure extension - "folderName" in destination data Address.
(https://github.com/eclipse-edc/Technology-Azure/blob/main/extensions/common/azure/azure-blob-core/src/main/java/org/eclipse/edc/azure/blob/AzureBlobStoreSchema.java
https://github.com/eclipse-edc/Technology-Azure/blob/main/extensions/data-plane/data-plane-azure-storage/src/main/java/org/eclipse/edc/connector/dataplane/azure/storage/pipeline/AzureStorageDataSink.java)
which then can be use in both cases - single and multi object transfer.

Reply:
Exactly, `keyPrefix` or `folderName` would do the job.

Reply:
Implemented in https://github.com/eclipse-edc/Technology-Aws/pull/156